client/extensions/woocommerce/state/wc-api/product-categories/reducer.js

```diff
@@ -0,0 +1,24 @@
+/**
+ * Internal dependencies
+ */
+import {
+	WOOCOMMERCE_API_PRODUCT_CATEGORIES_GET,
+	WOOCOMMERCE_API_PRODUCT_CATEGORIES_GET_SUCCESS,
+} from '../../action-types';
+
+export default {
+	[ WOOCOMMERCE_API_PRODUCT_CATEGORIES_GET ]: productCategoriesGet,
+	[ WOOCOMMERCE_API_PRODUCT_CATEGORIES_GET_SUCCESS ]: productCategoriesGetSuccess,
+};
+
+function productCategoriesGet( siteData ) {
+	// TODO: Update stats in the tree to show that this request is pending.
+	return siteData;
+}
+
+function productCategoriesGetSuccess( siteData, action ) {
+	const { data } = action.payload;
+
+	return { ...siteData, productCategories: data };
```

Comment:
Since the ultimate endpoint is untrusted code (on the merchant's host), we probably should consider validating the response before storing it in state.